src/chunk_dispatch_state.c

```diff
@@ -107,20 +147,32 @@ chunk_dispatch_exec(CustomScanState *node)
 			}
 
 			/* slot for the "existing" tuple in ON CONFLICT UPDATE IS chunk schema */
-
+#if PG12_LT
 			if (state->parent->mt_existing != NULL)
 			{
-				TupleDesc chunk_desc;
-
-				if (cis->tup_conv_map && cis->tup_conv_map->outdesc)
-					chunk_desc = cis->tup_conv_map->outdesc;
-				else
-					chunk_desc = RelationGetDescr(cis->rel);
 				Assert(chunk_desc != NULL);
 				ExecSetSlotDescriptor(state->parent->mt_existing, chunk_desc);
 			}
+#else
+			if (cis->result_relation_info->ri_onConflict != NULL)
+			{
+				if (cis->result_relation_info->ri_onConflict->oc_ProjSlot != NULL)
+				{
+					Assert(chunk_desc != NULL);
+					ExecSetSlotDescriptor(cis->result_relation_info->ri_onConflict->oc_ProjSlot, chunk_desc);
+				}
+
+				if (cis->result_relation_info->ri_onConflict->oc_Existing != NULL)
+				{
+					Assert(chunk_desc != NULL);
+					ExecSetSlotDescriptor(cis->result_relation_info->ri_onConflict->oc_Existing, chunk_desc);
+				}
+			}
+
+			ExecSetSlotDescriptor(state->slot, chunk_desc);
+#endif
 		}
-#if defined(USE_ASSERT_CHECKING) && PG11_GE
+#if defined(USE_ASSERT_CHECKING) && PG11
 		if (state->parent->mt_conflproj != NULL)
 		{
 			TupleTableSlot *slot = get_projection_info_slot_compat(
```

Comment:
`slot` is already declared at the beginning of the function.

Reply:
Since this seems to predate my changes, I'm going to hold off editting this unless it breaks PG11